js/models/messages.js

```diff
@@ -415,24 +415,49 @@
     },
 
     async acceptFriendRequest() {
+      const primaryDevicePubKey = this.attributes.conversationId;
+
       if (this.get('friendStatus') !== 'pending') {
         return;
       }
-      const conversation = await this.getSourceDeviceConversation();
-      // If we somehow received an old friend request (e.g. after having restored
-      // from seed, we won't be able to accept it, we should initiate our own
-      // friend request to reset the session:
-      if (conversation.get('sessionRestoreSeen')) {
-        conversation.sendMessage('', null, null, null, null, {
-          sessionRestoration: true,
-        });
-        return;
+
+      const allDevices = await libloki.storage.getAllDevicePubKeysForPrimaryPubKey(
+        primaryDevicePubKey
+      );
+
+      let profileName;
+      const allConversationsWithUser = allDevices.map(d =>
+        ConversationController.get(d)
+      );
+      allConversationsWithUser.forEach(conversation => {
+        // If we somehow received an old friend request (e.g. after having restored
+        // from seed, we won't be able to accept it, we should initiate our own
+        // friend request to reset the session:
+        if (conversation.get('sessionRestoreSeen')) {
+          conversation.sendMessage('', null, null, null, null, {
+            sessionRestoration: true,
+          });
+          return;
+        }
+
+        profileName = conversation.getProfileName() || profileName;
+        conversation.onAcceptFriendRequest();
+      });
+
+      // If you don't have a profile name for this device, and profileName is set,
+      // add profileName to conversation.
+      const primaryConversation = allConversationsWithUser.find(
+        c => c.id === primaryDevicePubKey
+      );
+      if (!primaryConversation.getProfileName() && profileName) {
+        await primaryConversation.setNickname(profileName);
       }
-      this.set({ friendStatus: 'accepted' });
+
       await window.Signal.Data.saveMessage(this.attributes, {
         Message: Whisper.Message,
       });
-      conversation.onAcceptFriendRequest();
+
+      this.set({ friendStatus: 'accepted' });
```

Comment:
Should'nt we trigger a redux update here?
with         `window.Signal.Data.updateConversation`